examples/sandbox/src/pages/customizing-styles-and-html/index.js

```diff
@@ -199,120 +123,11 @@ const config = {
-        {(contextProps) => (
-          <div className="App customization-example">
-            <ErrorBoundary>
-              <Layout
-                header={
-                  <SearchBox
-                    autocompleteMinimumCharacters={3}
-                    autocompleteResults={{
-                      linkTarget: "_blank",
-                      sectionTitle: "Results",
-                      titleField: "title",
-                      urlField: "nps_link",
-                      shouldTrackClickThrough: true,
-                      clickThroughTags: ["test"]
-                    }}
-                    autocompleteSuggestions={true}
-                    debounceLength={0}
-                  />
-                }
-                sideContent={
-                  <div>
-                    <ClearFilters />
-                    <br />
-                    <br />
-                    {contextProps.wasSearched && (
-                      <Sorting label={"Sort by"} sortOptions={SORT_OPTIONS} />
-                    )}
-                    <Facet
-                      field="states"
-                      label="States"
-                      filterType="any"
-                      isFilterable={true}
-                    />
-                    <Facet
-                      field="world_heritage_site"
-                      label="World Heritage Site?"
-                      view={BooleanFacet}
-                    />
-                    <Facet
-                      field="visitors"
-                      label="Visitors"
-                      view={SingleLinksFacet}
-                    />
-                    <Facet
-                      field="date_established"
-                      label="Date Established"
-                      filterType="any"
-                    />
-                    <Facet field="location" label="Distance" filterType="any" />
-                    <Facet
-                      field="acres"
-                      label="Acres"
-                      view={SingleSelectFacet}
-                    />
-                  </div>
-                }
-                bodyContent={
-                  <Results
-                    resultView={CustomResultView}
-                    titleField="title"
-                    urlField="nps_link"
-                    thumbnailField="image_url"
-                    shouldTrackClickThrough={true}
-                  />
-                }
-                bodyHeader={
-                  <>
-                    {contextProps.wasSearched && (
-                      <PagingInfo view={CustomPagingInfoView} />
-                    )}
-                    {contextProps.wasSearched && <ResultsPerPage />}
-                  </>
-                }
-                bodyFooter={<Paging />}
-              />
-            </ErrorBoundary>
-          </div>
-        )}
+        {(contextProps) => <CustomizingStylesApp contextProps={contextProps} />}
```

Comment:
More like this
```
<SearchProvider config={config}>
        <CustomizingStylesApp />
```
And inside CustomizingStylesApp use useSearch to getting context params

Reply:
if I understand correctly, `SearchContext` states are currently managed by `withSearch`, for this to work I think we will need to move logic in withSearch to another hook.

Reply:
updated in [commit](https://github.com/elastic/search-ui/pull/1130/commits/9949d54ad65608fe46565b59f57739db346efec8#diff-1ba05573ad68e69b0d6e1f496b5d1ef07588f56ddf0ae68893c454d371ab11b3)